src/apps/queues/views.py

```diff
@@ -0,0 +1,76 @@
+# from braces.views import LoginRequiredMixin
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.core.exceptions import PermissionDenied, ValidationError
+# from django.core.urlresolvers import reverse
+from django.db.models import Q
+from django.urls import reverse
+from django.forms.utils import ErrorList
+from django.http import HttpResponse
+from django.http import HttpResponseRedirect
+from django.views.generic import ListView, CreateView, UpdateView, DeleteView
+from pyrabbit.http import HTTPError
+
+from .forms import QueueForm
+from .models import Queue
+from . import rabbit
+
+
+class QueueFormMixin(object):
+    def get_success_url(self):
+        return reverse('queues:list')
+
+    def get_form_kwargs(self, *args, **kwargs):
+        kwargs = super(QueueFormMixin, self).get_form_kwargs(*args, **kwargs)
+        kwargs['user'] = self.request.user
+        return kwargs
+
+
+class QueueListView(LoginRequiredMixin, ListView):
+    model = Queue
+    template_name = 'queues/list.html'
+
+    def get_queryset(self):
+        qs = Queue.objects.filter(owner=self.request.user) | self.request.user.organized_queues.all()
+        return qs
+
+
+class QueueCreateView(LoginRequiredMixin, QueueFormMixin, CreateView):
+    model = Queue
+    template_name = 'queues/form.html'
+    form_class = QueueForm
+
+    def form_valid(self, form):
+        try:
+            queue = form.save(commit=False)
+            queue.owner = self.request.user
+            if rabbit.check_user_needs_initialization(self.request.user):
+                rabbit.initialize_user(self.request.user)
+            queue.vhost = rabbit.create_queue(self.request.user)
+
+            # Only save queue if things were successful
+            queue.save()
+            form.save_m2m()
```

Comment:
maybe add small comment here that this is for saving the ManyToMany relationships from the form? since we don't use forms super often, a little extra commenting will make maintaining this nicer?